mspnp/aks-baseline-regulated
# feat (workload identity): [aks-ingress] migrate pod to workload identity

closes: #24723

## WHY?

we do want to move all our aks baseline from aad pod identity manually installed or via add-ons to workload identity.

## WHAT changed?

- migrate from AAD Pod Identity to Azure Workload Identity (Add on)
- minor bug fixings 

## HOW to test?

as a user you can now clone the aks regulated baseline repository, follow the instructions and the cluster stamp v2 will create federated credentials identity subject to the nginx ingress controller service account and enable the workload identity add on in the aks regulated cluster.

you can deploy the cluster end to end and validate it is working as expected.
